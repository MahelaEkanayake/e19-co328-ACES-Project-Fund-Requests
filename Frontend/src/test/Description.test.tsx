

import { render, screen } from '@testing-library/react';
import Description from '../components/Description'; // Adjust the import path as necessary

describe('Description Component', () => {
  it('renders the ACES Project Fund text', () => {
    render(<Description />);
    const headingElement = screen.getByText(/ACES Project Fund/i);
    expect(headingElement).toBeTruthy();
  });

  it('renders the application form heading', () => {
    render(<Description />);
    const formHeadingElement = screen.getByText(/Application Form/i);
    expect(formHeadingElement).toBeTruthy();
  });

  it('renders the project fund description', () => {
    render(<Description />);
    const descriptionText = /The project fund is set up by ACES with the help of donations from ACES Alumni./i;
    const descriptionElement = screen.getByText(descriptionText);
    expect(descriptionElement).toBeTruthy();
  });
});

