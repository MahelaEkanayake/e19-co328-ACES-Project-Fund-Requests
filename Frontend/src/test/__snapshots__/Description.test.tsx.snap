// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description component matches snapshot 1`] = `
<div>
  <div
    class="css-1jq9h04"
  >
    <p
      class="chakra-text css-1oacvqt"
    >
      ACES Project Fund
    </p>
    <div
      class="css-5qr6tg"
    >
      <p
        class="chakra-text css-1mnnhnb"
      >
        The project fund is set up by ACES with the help of donations from ACES Alumni. It is managed by a committee appointed by ACES and the members are ACES executives and the Head of the Department.
      </p>
    </div>
    <p
      class="chakra-text css-ca6j6m"
    >
      Application Form
    </p>
  </div>
</div>
`;
